Gemfile

```diff
@@ -8,15 +8,8 @@ gem 'onceover',                                                  '~> 3.20', '>=
 gem 'onceover-codequality',                                      '~> 0.8', '>= 0.8.0'
 gem 'onceover-octocatalog-diff',                                 '~> 0.1', '>= 0.1.8'
 gem 'onceover-lookup',                                           '~> 0.1', '>= 0.1.1'
-gem 'parallel',                                                  '~> 1.20', '>= 1.20.1'               
-gem 'puppet-lint-classes_and_types_beginning_with_digits-check', '~> 1.0'
-gem 'puppet-lint-leading_zero-check',                            '~> 1.0'
-gem 'puppet-lint-legacy_facts-check',                            '~> 1.0', '>= 1.0.4'
-gem 'puppet-lint-top_scope_facts-check',                         '~> 1.0'
-gem 'puppet-lint-trailing_comma-check',                          '~> 0.4.1'
-gem 'puppet-lint-unquoted_string-check',                         '~> 2.0'
-gem 'puppet-lint-variable_contains_upcase',                      '~> 1.2'
-gem 'puppet-lint-version_comparison-check',                      '~> 1.0'
+gem 'parallel',                                                  '~> 1.20', '>= 1.20.1'
+gem 'voxpupuli-puppet-lint-plugins',                             '~> 3.1'
```

Comment:
we're currently preparing the 4.0 release. it should be out today.

Reply:
will update then.

Reply:
@bastelfreak when i also update puppet-lint to 3.1 as the voxpupuli-lint-plugins require with version 4.0 i run into the following:

```
➜  puppet-dev-tools git:(update_gems) ✗ bundle install --path vendor/bundle
Fetching gem metadata from https://rubygems.org/.........
Fetching gem metadata from https://rubygems.org/.
Resolving dependencies...
Bundler could not find compatible versions for gem "puppet-lint":
  In Gemfile:
    puppet-lint (~> 3.1)

    onceover-codequality (~> 0.8) was resolved to 0.8.0, which depends on
      puppet-lint (~> 2)

Bundler could not find compatible versions for gem "puppet-syntax":
  In Gemfile:
    puppet-syntax (~> 3.2.0)

    puppetlabs_spec_helper (~> 5.0) was resolved to 5.0.3, which depends on
      puppet-syntax (~> 3.0)

Bundler could not find compatible versions for gem "rubocop":
  In Gemfile:
    rubocop (~> 1.22.3)

    rubocop-performance (~> 1.16.0) was resolved to 1.16.0, which depends on
      rubocop (>= 1.7.0, < 2.0)

    rubocop-rspec (~> 2.11.1) was resolved to 2.11.1, which depends on
      rubocop (~> 1.19)

Bundler could not find compatible versions for gem "ruby":
  In Gemfile:
    ruby

    nokogiri (~> 1.10, >= 1.10.9) was resolved to 1.14.2, which depends on
      ruby (>= 2.7.0)
```

Reply:
you're on Ruby 2.5 I assume? This is intentional, puppet-strings 3 dropped support for Ruby 2.5/2.6. At Vox Pupuli we've one CI job, running on Ruby 2.7, to to basic code verification, REFERENCE.md check, linting. That runs with puppet-strings 3. Afterwards we generate Ruby/Puppet version specific CI jobs, without puppet-strings 3.

Reply:
Ah I think `onceover-codequality` doesn't support puppet-lint 3.

Reply:
onceover-codequality is now updated to 0.9. and now we can use all the new shiny gems
tested with ruby 2.7.7 on my local